hnn_core/network.py

```diff
@@ -764,18 +830,16 @@ def _instantiate_drives(self, n_trials=1):
                                 seedcore=drive['seedcore'])
                             )
                     else:
-                        # cell_specific=False should only have one src_gid
-                        assert len(this_cell_drive_conn['src_gids']) == 1
-
                         # Only return empty event times if all cells have
                         # no events
                         drive_cell_gid = this_cell_drive_conn[
                             'src_gids'][0]
-                        event_times = [_drive_cell_event_times(
+                        event_times = ([_drive_cell_event_times(
                             drive['type'], this_cell_drive_conn,
                             drive['dynamics'], trial_idx=trial_idx,
                             drive_cell_gid=drive_cell_gid,
-                            seedcore=drive['seedcore'])]
+                            seedcore=drive['seedcore'])] *
+                            len(this_cell_drive_conn['src_gids']))
```

Comment:
break into two or three lines for readability?

```py
n_drive_cells =  len(this_cell_drive_conn['src_gids']
event_times_cell = _drive_cell_event_times( ...)
event_times = [event_times_cell] * n_drive_cells

```

Reply:
This whole section needs some help given the recent changes. On a side note, why does `_drive_cell_event_times()` depend on the specific connectivity pattern of a given source cell type?

Reply:
it shouldn't. I think you should only rely on `net.gid_ranges['drivename']`. This is part of what I stumbled into in #369 . Maybe just try changing and seeing if something breaks?